설정/회원가입

xwoud opened the pull request:
## 🤗 What is this PR?

- 미회원일 때 회원가입 후 닉네임 정하는 View
- issue: #33
- <img width="1077" alt="스크린샷 2022-01-10 오후 11 53 47" src="https://user-images.githubusercontent.com/51286963/148786296-d5914645-830d-4c6b-816b-3046648fc86e.png">


## 📸 Screenshot

iphoneSE2 | iphone13| iphone13ProMax  |
:---: | :---: | :---:
<img width="250" src="https://user-images.githubusercontent.com/51286963/148786395-16c409a9-ba7f-493e-bc5d-3b047e54d913.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148786476-71c88a3a-454f-4787-a97f-0f056ceaa1ab.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148786485-ae71f242-5211-422b-8794-cab9842cca16.png">
<img width="250" src="https://user-images.githubusercontent.com/51286963/148788556-48f1c80f-6e1c-48f7-8b25-2f8a3f525398.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148788568-beedf05b-eac9-4a9d-941f-f4703a2f1d27.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148788576-ca00d3fe-22d9-4ef9-b8f5-1ddc94c62a5f.png">
<img width="250" src="https://user-images.githubusercontent.com/51286963/148788745-c622c05f-9b09-4edb-9944-741f2f29f055.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148788763-684aa8d0-757d-4662-a8b0-4c04c7c084ba.png"> | <img width="250" src="https://user-images.githubusercontent.com/51286963/148788774-bbd3cdc3-7c1d-4d9a-bacf-9a99eca5a60d.png">

## ✅ Test Checklist

- [x] 다 지웠을 때 placeholder 나오기
- [x] '확인'버튼 클릭 시 중복 누르면 중복 라벨 나오기
- [x] 글자수 6자 이상 시 '확인' 버튼 비활성화
- [x] 닉네임에 한글, 영어, _ 만 사용 가능

## 💬 Comment
- 조건 맞는지 봐주세요
- 현재 '카카오톡 로그인' 버튼만 연결해둠
<img width="500" alt="스크린샷 2022-01-11 오전 12 13 02" src="https://user-images.githubusercontent.com/51286963/148789480-f1fd564b-b026-40f6-bbb0-ecd1149f4c89.png">

xwoud committed:
[:lipstick: layout] 회원가입 뷰 storyboard object 배치

xwoud committed:
[:lipstick: layout] 회원가입 뷰 디테일한 레이아웃 생성

feat: #33

xwoud committed:
[:wrench: configure] 회원가입 후 닉네임 정하는 뷰 생성

feat: #33

xwoud committed:
[:lipstick: layout] 닉네임 정하기 뷰 레이아웃 잡기 with Storyboard

feat: #33

xwoud committed:
[:sparkles: feature] 닉네임 글자수 중복 검사 및 문자 제한

feat: #33

xwoud committed:
[:sparkles: feature] 닉네임 검사 후 버튼 활성화 및 비활성화 기능 추가

feat: #33

xwoud committed:
[:lipstick: layout] 중복 검사 후 중복 문구 등장

feat: #33

xwoud committed:
[:lipstick: layout] 닉네임 조건 변경

feat: #33

xwoud committed:
[:lipstick: layout] 확인 버튼 크기 수정

feat: #33

xwoud committed:
Merge branch 'develop' into feature/#33

Juhyeoklee commented on Code-Zero/Screen/SignUp/VC/NickSettingVC.swift:
```diff
@@ -0,0 +1,112 @@
+            } else { // 중복 아닐 시
+                // 설정 페이지로 이동
+            }
+        }
```
guard let 으로 indent를 줄여보는 건 어떨까요 ?!

Juhyeoklee commented on Code-Zero/Screen/SignUp/VC/NickSettingVC.swift:
```diff
@@ -0,0 +1,112 @@
+        nickBoxView.setBorder(borderColor: .gray2, borderWidth: 1)
+        duplicateCheckLabel.isHidden = true
+        nickTextField.delegate = self
+        nickTextField.inputAccessoryView = accessoryView
```
오호 accessoryView로 넣어주면 따로 노티를 이용하지 않아도 키보드 애니메이션이 되는건가요 ??

xwoud replied:
accessoryView로 하면 기냥 키보드 바로 위에 찰싹 붙어서 따로 키보드가 올라갈 때 저 버튼을 올려주거나 하지 않아두되여~~

Juhyeoklee replied:
정말 멋진 방법이네용 ~~

Juhyeoklee commented on Code-Zero/Screen/SignUp/VC/NickSettingVC.swift:
```diff
@@ -0,0 +1,112 @@
+    @objc func textFieldDidChange(_ textField: UITextField) {
+        duplicateCheckLabel.isHidden = true
+        textField.text = textField.text?.trimmingCharacters(in: .whitespaces) // 스페이스 제거
+        if let text = textField.text {
+            if let regexText = filterNickName(text: text) {
+                textField.text = regexText
+            }
+        }
```
요쪽에서 버그가 발생하는것 같습니당 허용안되는 특수문자를 세번 연속으로 입력할 때 입력한 텍스트가 지워져요

```swift
textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String)
```
UITextFieldDelegate의 위 메소드를 활용해 보는게 어떨까요 ?

xwoud replied:
이것은 한글 글자수 체크에 문제가 생겨서... 잠만 보류

Juhyeoklee reviewed (commented):
LGTM :) 😊

xwoud committed:
[:recycle: refactor] guard let을 이용해 indent 줄이기

xwoud committed:
[:poop: poop] 닉네임 글자수 제한 똥쌌습니다

feat: #33

xwoud committed:
[style] 닉네임 변경 수정

xwoud committed:
[:lipstick: layout] 닉네임 조건 설정 화면 변경

body: 5글자 제한 글자 및 7자 이상 쳐지기 금지

feat: #33

xwoud committed:
Merge branch 'develop' into feature/#33

xwoud commented:
### 닉네임 조건 변경 

- 0자 작성 시 
     - placdholder 보이기
     - 버튼 비활성화
     - 알럿 문구 X
- 1자 ~ 5자 작성 시
      - 버튼 활성화
      - 알럿 문구 X
- 5자 초과 ~ 7자
     - 버튼 비활성화
     - 알럿 문구 O
- 7자 초과 입력 불가

### 안되는 점
- 자꾸 닉네임에 -- 이렇게 두번치면 한글자 지워짐 ㅠ.ㅠ 글자 필터좀 봐주세요

xwoud committed:
[:zap: improve] 문자 제한 기능 업그레이드

body: shouldChangeCharactersIn에서 글자의 특수문자 제한
만약 특수문자 제한을 통과한다면 글자수 검사도 진행

feat: #33
